main/src/cgeo/geocaching/ui/CalculateButton.java

```diff
@@ -0,0 +1,267 @@
+            case INPUT_VAL: rv = buttonData.inputVal;   break;
+            case AUTO_CHAR: rv = buttonData.autoChar;   break;
+            case CUSTOM:    rv = buttonData.customChar; break;
+            default:        rv = '*'; // Should never happen.
+        }
+
+        return rv;
+    }
+
+    public void setLabel(final char theLabel) {
+        butt.setText(String.valueOf(theLabel));
+    }
+
+    public ValueType getType() {
+        return buttonData.type;
+    }
+
+    public void setType(final ValueType theType) {
+        buttonData.type = theType;
+        setBackgroundColour();
+    }
+
+    // Buttons displaying a custom value are given a 'steel-like' appearance so as to distinguish them from regular buttons.
+    private  void setBackgroundColour() {
+        if (buttonData.type == ValueType.CUSTOM) {
+            butt.setBackgroundColor(ContextCompat.getColor(getContext(), R.color.button_default));
+            butt.setTextColor(ContextCompat.getColor(getContext(), R.color.steel));
+            butt.setTypeface(null, Typeface.BOLD);
+        } else {
+            butt.setBackgroundResource(R.drawable.button_background_kitkat);
+            butt.setTextColor(Color.WHITE);
+            butt.setTypeface(null, Typeface.NORMAL);
+        }
+    }
+
+    // Data used to preserve this buttons state between sessions.
+    public ButtonData getData() {
+        return buttonData;
+    }
+
+    public void setData(final ButtonData buttonData) {
+        this.buttonData = buttonData;
+        setBackgroundColour();
+        updateButtonText();
+    }
+
+    public void setInputVal(final String theVal) {
+        setInputVal(theVal.charAt(0));
+    }
+
+    public void setInputVal(final char theVal) {
+        buttonData.inputVal = theVal;
+        updateButtonText();
+    }
+
+    public CalculateButton getNextButton() {
+        return nextButton;
+    }
+
+    public void setNextButton(final CalculateButton theNext) {
+        nextButton = theNext;
+    }
+
+    /**
+     * This method is used to assign buttons alphabetically increasing names.
+     * @param currentChar:  values indicated that the autoChar will be and is updated appropriately for the next button based on the current button's state.
+     */
+    public void setAutoChar(final char currentChar) {
+        char nextChar = currentChar;
+
+        if (nextChar < 'A') {
+            nextChar = 'A';
+        } else if (nextChar > 'Z') {
+            nextChar = 'Z';
+        }
+
+        switch (getType()) {
+            case INPUT_VAL:
+                buttonData.autoChar = nextChar;
+                break;
+
+            case AUTO_CHAR:
+                buttonData.autoChar = nextChar++;
+                break;
+
+            case CUSTOM:
+                buttonData.autoChar = nextChar;
+                if ('A' <= buttonData.customChar && buttonData.customChar <= 'Z') {
+                    nextChar = buttonData.customChar;
+                    nextChar++;
```

Comment:
The customChar could be either a letter, a number, or the space character.  The idea of the range check is to ensure customChars only affect the ordering of the autoChars when they are set to be a letter.

While it's true that both the digits and the space character have ascii values less than that of the letters it seemed appropriate to set the upper bound as well rather than to rely upon the restriction of the keyboard input.  And yes, a 'Z' will be followed by more 'Z's but even a custom 'Z' should affect the next autoChar.